lib/icon/module.js

```diff
@@ -45,6 +46,13 @@ async function run (pwa, _emitAssets) {
     return
   }
 
+  // Verify purpose
+  if (options.purpose) {
+    const purpose = Array.isArray(options.purpose) ? options.purpose : [options.purpose]
+    const validPurpose = ['badge', 'maskable', 'any']
+    options.purpose = purpose.filter(item => validPurpose.includes(item))
```

Comment:
Silently filtering invalid items is not a good idea. Maybe we can show a warning instead? (one idea can be keeping length before filter and warn if length changed)

Reply:
warn user is a good idea

Reply:
0ff11d41434daa3cb37459606c82b1a5fbdeb293